packages/flutter/lib/src/material/refresh_indicator.dart

```diff
@@ -215,12 +215,21 @@ class RefreshIndicatorState extends State<RefreshIndicator> with TickerProviderS
     super.dispose();
   }
 
+  bool _shouldStart(ScrollNotification notification) {
+    // The indicator will be pulled out in two cases,
+    // 1, Begin drag when the scrollable widget at the edge with zero scroll position.
+    // 2, Keep drag before overscroll occurs when the scrollable widget has
+    //    a non-zero scroll position(do not release finger before overscroll).
+    return (notification is ScrollStartNotification || (notification is ScrollUpdateNotification && notification.dragDetails != null))
```

Comment:
Since you have refactor out this check, can you also implement the opt in behavior as well? We can have an widget parameter to control whether it can be pull out if it drag from non zero scroll offset?

Reply:
OK, BUT I still want to talk to @Hixie  about this behavior. I want to know what he is thinking right now. I don't think the refresh indicator is easily pulled out by misoperation.It must continue to pull down a certain distance before it can be refreshed. What do you think,?